### ISSUE TITLE
Align axis scaling in subplots

### ISSUE THREAD

**Issue by hoefling** [altair-viz/altair#2390]
First of all, this library is so awesome. Found via [this PyCon talk](https://www.youtube.com/watch?v=vTingdk_pVM), using for the first time and already in love with it!

As an `altair` newbie, I have encountered the following issue: when plotting two chart sets over the same dataset with the same x axis data, the axes have different bounds. The data:
```python
In [5]: df.describe()
Out[5]: 
       learning_rate         alpha           MSE          RMSE      epochs
min         0.000001  1.000000e-10  2.535845e-15  5.035717e-08    0.000000
max         0.100000  6.309573e-01  5.826400e-01  7.633086e-01  313.000000
```
As you can see, `alpha` is in `[1e-10, 0.63]`.

A minimal chart example boils down to:

```python
log_alpha = alt.X("alpha", scale=alt.Scale(type="log"))

mse_chart = alt.Chart(df).mark_point().encode(
    x=log_alpha,
    y=alt.Y("MSE", scale=alt.Scale(type="log")),
    column="learning_rate",
)
epochs_chart = alt.Chart(df).mark_bar().encode(
    x=log_alpha,
    y="epochs:Q",
    column="learning_rate",
)

(mse_chart & epochs_chart).save("chart.html")
```

The rendered result: https://hoefling.io/chart. As you can see, the scatter plots have the x axis bound `[1e-10, 1]`, while the bar plots have `[1e-11, 1]`. Is there some setting I can use to align the axes? I see both [`Chart.resolve_axis()`](https://altair-viz.github.io/user_guide/generated/toplevel/altair.Chart.html#altair.Chart.resolve_axis) and [`Chart.resolve_scale`](https://altair-viz.github.io/user_guide/generated/toplevel/altair.Chart.html#altair.Chart.resolve_scale), but unsure how to apply them correctly and the docstrings don't help.

**Comment by jakevdp** [comment]
Can you include a complete minimal example that demonstrates the issue?

**Comment by hoefling** [comment]
@jakevdp gladly:

<details>
  <summary><code>altair-ex.py</code></summary>

```python
import io
import pathlib
import altair as alt
import pandas as pd


def main(df):
    log_alpha = alt.X("alpha", scale=alt.Scale(type="log"))
    mse_chart = alt.Chart(df).mark_point().encode(
        x=log_alpha,
        y=alt.Y("MSE", scale=alt.Scale(type="log")),
        column="learning_rate",
    )
    epochs_chart = alt.Chart(df).mark_bar().encode(
        x=log_alpha,
        y="epochs:Q",
        column="learning_rate",
    )
    # store as self-embedding HTML
    (mse_chart & epochs_chart).save("/tmp/chart.html")


def vega_lite_4680():
    # https://github.com/vega/vega-lite/issues/4680#issuecomment-749177420
    html_file = pathlib.Path("/tmp/chart.html")
    html = html_file.read_text()
    html = html.replace(
        '"vconcat": [{"mark":',
        '"vconcat": [{"transform": [{"filter": "(datum.alpha >= -1e-10)"}], "mark":',
        1
    )
    pathlib.Path(html_file).write_text(html)


raw = """
,learning_rate,alpha,trained,MSE,RMSE,epochs
0,0.00215625,0.39810717,True,0.011021161600481718,0.10498172039623284,313
1,0.00215625,0.39810717,False,0.18499652464687824,0.43011222330303744,0
2,1e-06,1e-05,True,2.049491863304054e-10,1.4316046458954417e-05,313
3,1e-06,1e-05,False,1.9269387644405e-09,4.3896910676316964e-05,0
4,1e-05,0.0001,True,1.3561040620224673e-08,0.00011645188118164302,313
5,1e-05,0.0001,False,1.9244220317204964e-07,0.0004386823490392488,0
6,0.00215625,1e-09,True,2.536635202700481e-15,5.03650196013399e-08,15
7,0.00215625,1e-09,False,2.5540587887983563e-15,5.053769669680264e-08,0
8,1e-05,0.001,True,1.351696892356813e-06,0.0011626250012390475,313
9,1e-05,0.001,False,1.8981234115017286e-05,0.004356745818277196,0
10,0.1,0.0001,True,5.2230779914877804e-12,2.285405424658925e-06,170
11,0.1,0.0001,False,1.9244220317204964e-07,0.0004386823490392488,0
12,0.001,0.001,True,6.879695564521172e-08,0.0002622917376086567,313
13,0.001,0.001,False,1.8981234115017286e-05,0.004356745818277196,0
14,0.00215625,1e-10,True,2.5358902513052937e-15,5.0357623528592475e-08,15
15,0.00215625,1e-10,False,2.5358450712031617e-15,5.035717489993615e-08,0
16,1e-05,1e-05,True,1.3565587867608864e-10,1.1647140363511359e-05,313
17,1e-05,1e-05,False,1.9269387644405e-09,4.3896910676316964e-05,0
18,0.0001,1e-06,True,5.836264320132137e-13,7.639544699198575e-07,313
19,0.0001,1e-06,False,1.927375173176038e-11,4.390188126410356e-06,0
20,0.01,1e-06,True,8.149108348451379e-15,9.027241189741117e-08,313
21,0.01,1e-06,False,1.927375173176038e-11,4.390188126410356e-06,0
22,0.00215625,1e-08,True,2.5630034152167472e-15,5.06261140004231e-08,36
23,0.00215625,1e-08,False,4.4533639333769845e-15,6.673352941312332e-08,0
24,0.01,0.001,True,5.630914039267409e-09,7.503941660215516e-05,313
25,0.01,0.001,False,1.8981234115017286e-05,0.004356745818277196,0
26,1e-05,1e-06,True,1.3587936424901831e-12,1.1656730416004026e-06,313
27,1e-05,1e-06,False,1.927375173176038e-11,4.390188126410356e-06,0
28,0.001,0.0001,True,6.862863419221376e-10,2.6197067461664296e-05,313
29,0.001,0.0001,False,1.9244220317204964e-07,0.0004386823490392488,0
30,0.00215625,0.15848932,True,0.0017978303026175127,0.04240082904576833,313
31,0.00215625,0.15848932,False,0.06523100837692618,0.2554036187986022,0
32,0.00215625,0.25118864,True,0.0052517254077829425,0.07246878916893751,313
33,0.00215625,0.25118864,False,0.09679379159212112,0.31111700628122124,0
34,0.1,1e-06,True,3.249159871043313e-15,5.7001402357415086e-08,134
35,0.1,1e-06,False,1.927375173176038e-11,4.390188126410356e-06,0
36,0.001,0.01,True,7.060038055897166e-06,0.0026570732123910143,313
37,0.001,0.01,False,0.0016602828385075554,0.04074656842742723,0
38,1e-06,0.0001,True,2.048715772673404e-08,0.00014313335643627797,313
39,1e-06,0.0001,False,1.9244220317204964e-07,0.0004386823490392488,0
40,0.00215625,0.63095734,True,0.015091352353803813,0.12284686564120735,71
41,0.00215625,0.63095734,False,0.5826399919390678,0.7633085820254111,0
42,0.00215625,1e-05,True,3.5640372339582754e-12,1.8878657891217e-06,313
43,0.00215625,1e-05,False,1.9269387644405e-09,4.3896910676316964e-05,0
44,1e-06,0.001,True,2.0413193873309866e-06,0.0014287474890648205,313
45,1e-06,0.001,False,1.8981234115017286e-05,0.004356745818277196,0
46,0.001,1e-06,True,7.112600381237112e-14,2.666945889124539e-07,313
47,0.001,1e-06,False,1.927375173176038e-11,4.390188126410356e-06,0
48,0.00215625,0.0001,True,3.5618442800755547e-10,1.887284895897684e-05,313
49,0.00215625,0.0001,False,1.9244220317204964e-07,0.0004386823490392488,0
50,0.0001,0.01,True,5.7868745250743816e-05,0.007607150931649257,313
51,0.0001,0.01,False,0.0016602828385075554,0.04074656842742723,0
52,1e-05,0.01,True,0.00013038527058233739,0.011418636979368542,313
53,1e-05,0.01,False,0.0016602828385075554,0.04074656842742723,0
54,1e-06,0.01,True,0.00019582369369163644,0.013993701930498719,313
55,1e-06,0.01,False,0.0016602828385075554,0.04074656842742723,0
56,0.0001,0.0001,True,5.813607678817334e-09,7.624701756576428e-05,313
57,0.0001,0.0001,False,1.9244220317204964e-07,0.0004386823490392488,0
58,0.00215625,1e-07,True,3.087116472616727e-15,5.556182565574036e-08,190
59,0.00215625,1e-07,False,1.9516304199075023e-13,4.4177261359553554e-07,0
60,0.01,1e-05,True,5.641224898100599e-13,7.510808814920768e-07,313
61,0.01,1e-05,False,1.9269387644405e-09,4.3896910676316964e-05,0
62,0.01,0.0001,True,5.619449049663883e-11,7.496298452559574e-06,313
63,0.01,0.0001,False,1.9244220317204964e-07,0.0004386823490392488,0
64,0.1,0.001,True,5.238289079931291e-10,2.2887308901493915e-05,170
65,0.1,0.001,False,1.8981234115017286e-05,0.004356745818277196,0
66,0.0001,1e-05,True,5.8137981946010384e-11,7.6248266794331465e-06,313
67,0.0001,1e-05,False,1.9269387644405e-09,4.3896910676316964e-05,0
68,0.00215625,1e-06,True,3.814943109000822e-14,1.9531879349094585e-07,313
69,0.00215625,1e-06,False,1.927375173176038e-11,4.390188126410356e-06,0
70,0.00215625,0.1,True,0.0006363679512869566,0.0252263344674445,313
71,0.00215625,0.1,False,0.044331937570124866,0.21055150813848922,0
72,1e-06,1e-06,True,2.051615898977961e-12,1.432346292817456e-06,313
73,1e-06,1e-06,False,1.927375173176038e-11,4.390188126410356e-06,0
74,0.00215625,0.001,True,3.568501200668095e-08,0.00018890476948677196,313
75,0.00215625,0.001,False,1.8981234115017286e-05,0.004356745818277196,0
76,0.1,1e-05,True,5.4503111300162246e-14,2.3345901443977498e-07,170
77,0.1,1e-05,False,1.9269387644405e-09,4.3896910676316964e-05,0
78,0.00215625,0.01,True,3.640025068079922e-06,0.001907884970506822,313
79,0.00215625,0.01,False,0.0016602828385075554,0.04074656842742723,0
80,0.0001,0.001,True,5.811538786417714e-07,0.00076233449248666,313
81,0.0001,0.001,False,1.8981234115017286e-05,0.004356745818277196,0
82,0.001,1e-05,True,6.863569271525066e-12,2.6198414592894905e-06,313
83,0.001,1e-05,False,1.9269387644405e-09,4.3896910676316964e-05,0
"""

if __name__ == "__main__":
    df = pd.read_csv(io.StringIO(raw))
    main(df)
    vega_lite_4680()
```
</details>

To reproduce: `python altair-ex.py && open /tmp/chart.html` on MacOS, or `python altair-ex.py && xdg-open /tmp/chart.html` on Linux etc.

**Comment by hoefling** [comment]
@jakevdp it looks to me like it's `vega-lite`'s issue rather than `altair`'s. [Gist in Vega Editor that reproduces that](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAVhDYA7EABoQAEzjQATjRyZ289AEEABBBoIcguIaZJ1h2DcyGA7nRiHETOMtXLDypJhUiioBKKigxEiCDGpoANqgYQCMcPxJAAxKTGgATAAcAL4KiWgp-ACcWWgArNVFJahleVWoYgDM9SDJqQDsLeVJnWHplOljY0kteYPFXWgj4+mTINmo1R2zw6MTLUnlQ-Pby6t5PQeoIwBsLdU5BQC6RSDI6gDWaCs2wXCyUGzKNFkZDQoAAHiCQAAzGhwQTKD4oJSYACeODgHwAjgwkLIdIEdKRgtAoujwiAUWiPoI2MCCk9kRDobD4ehskjUaSQFicXiAjRCXSCkA). I managed to overcome this by using separate `alt.X` instances for both charts and adjusting the axis domains, e.g.
```py
log_alpha = alt.X("alpha:Q", scale=alt.Scale(type="log", domain=(1e-11, 1)))
bar_log_alpha = alt.X("alpha:Q", scale=alt.Scale(type="log"))

mse_chart = alt.Chart(df).mark_circle(clip=True).encode(
    x=log_alpha,
    y=alt.Y("MSE:Q", scale=alt.Scale(type="log")),
    column="Learning rate",
)
epochs_chart = alt.Chart(df).mark_bar().encode(
    x=bar_log_alpha,
    y="Epochs:Q",
    column="Learning rate",
)
```
This will scale the scatter plot to `[1e-11, 1]`, thus matching the x axis scale of the bar chart. There's some white space in the scatter plots now, but honestly I couldn't care less. Thus closing this - sorry for the noise!

**Comment by jakevdp** [comment]
Glad you figured it out – I started trying to understand your example, but was confused by all the display boilerplate. Next time just include the code for the chart.
